src/Microsoft.NET.Sdk.Functions.Generator/FunctionJsonConverter.cs

```diff
@@ -173,36 +175,78 @@ private bool TryGenerateFunctionJsons()
                 AssemblyResolver = resolver
             };
 
-            var module = ModuleDefinition.ReadModule(_assemblyPath, readerParams);
+            loadContext.Resolving += (context, assemblyName) =>
+            {
+                Assembly result = null;
+
+                // First, check the assembly root dir. Assemblies copied with the app always wins.
+                var path = Path.Combine(assemblyRoot, assemblyName.Name + ".dll");
+                if (File.Exists(path))
+                {
+                    result = context.LoadFromAssemblyPath(path);
+
+                    if (result != null)
+                        return result;
+                }
+
+                // Then, check if the assembly is already loaded into the default context.
+                // This is typically the case for framework assemblies like System.Private.Corelib.dll.
+                if (AssemblyLoadContext.Default.Assemblies.FirstOrDefault(a => a.GetName().Equals(assemblyName)) is Assembly existing)
+                {
+                    path = existing.Location;
+                    result = context.LoadFromAssemblyPath(path);
+
+                    if (result != null)
+                        return result;
+                }
 
-            IEnumerable<TypeDefinition> exportedTypes = module.Types;
+                // Lastly, try the resolution logic used by cecil. This can produce assemblies with
+                // different versions, which may cause issues. But not doing this fails in more cases.
```

Comment:
Do you have a scenario where you hit this? I understand the two scenarios above, but not this one...

Reply:
Yes. Different versions of `System.Runtime`. The generator was running on `4.x.2.0` (going off memory here), and my project references `4.x.3.0`. Cecil module resolution gives me the `4.x.2.0` one which is technically not correct cause it's a lower version than what I requested, but it works for this case at least. An alternative would be inserting some logic to look in NuGet folders or something I guess. Also, do note the version numbers I stated above might be wrong, but still should show the issue. I was targeting a patch later than what the generator was using basically.

Reply:
I thought the logic above would catch that (where you try to find the assembly from the default context)? Or maybe that's looking for a specific version still? I always think `assemblyName` is just the name, but then I forget that it includes a specific version as well :-)

Yes, we don't really care what version of (most) assemblies get loaded -- we just need enough to get the Attribute created via reflection. And loading slightly older versions of these runtime assemblies is fine for that case.

Reply:
Fun fact: the project that's not building for me (which is why I made this PR) builds on our CI which only has 2 versions of the .NET SDK installed. So yeah, dealing with multiple versions of .NET framework assemblies is weird 😛.

Reply:
And yes, `AssemblyName` contains version. The equality check returns false.